src/api/space.py

```diff
@@ -0,0 +1,116 @@
+""" Space related functions """
+from decimal import Decimal
+
+import requests
+import logging
+from typing import Dict, Any
+from requests.exceptions import HTTPError, ConnectionError, Timeout, RequestException
+
+from src.constants.constants import SWAN_API, DEPLOYMENT_SPACE
+from src.exceptions.request_exceptions import (
+    SwanRequestError,
+    SwanTimeoutError,
+    SwanHTTPError,
+)
+from src.exceptions.swan_base_exceptions import SwanValueError
+
+
+def deploy_space_v1(
+    paid: Decimal,
+    duration: int,
+    cfg_name: str,
+    region: str,
+    start_in: int,
+    tx_hash: str,
+    job_source_uri: str,
+) -> Dict[str, Any]:
+    """
+    Deploy a space with given parameters.
+
+    This function sends a POST request to a space deployment API endpoint. It handles the necessary data formatting
+    and error checking.
+
+    Args:
+        paid (Decimal): The amount paid for the deployment.
+        duration (int): The duration of the deployment.
+        cfg_name (str): The name of the configuration.
+        region (str): The region for the deployment.
+        start_in (int): The start time in minutes.
+        tx_hash (str): The transaction hash.
```

Comment:
Please make a note in the comments that this transaction hash is for the **payment made** for the given service, we use this tx hash to cross reference with the paid amount